apps/ejabberd/src/mod_mam.erl

```diff
@@ -544,66 +570,68 @@ handle_set_message_form(
-    {error, Reason} ->
-        report_issue(Reason, mam_lookup_failed, ArcJID, IQ),
-        return_error_iq(IQ, Reason);
-    {ok, {TotalCount, Offset, MessageRows}} when IQ#iq.xmlns =:= ?NS_MAM_03 ->
-        ResIQ = IQ#iq{type = result, sub_el = []},
-        %% Server accepts the query
-        ejabberd_router:route(ArcJID, From, jlib:iq_to_xml(ResIQ)),
-
-        %% Forward messages
-        {FirstMessID, LastMessID} =
-            case MessageRows of
-                []    -> {undefined, undefined};
-                [_|_] -> {message_row_to_ext_id(hd(MessageRows)),
-                          message_row_to_ext_id(lists:last(MessageRows))}
-            end,
-        [send_message(ArcJID, From, message_row_to_xml(MamNs, set_client_xmlns_for_row(M), QueryID))
-         || M <- MessageRows],
-
-        %% Make fin message
-        IsLastPage = is_last_page(PageSize, TotalCount, Offset, MessageRows),
-        IsStable = true,
-        ResultSetEl = result_set(FirstMessID, LastMessID, Offset, TotalCount),
-        FinMsg = make_fin_message(IQ#iq.xmlns, IsLastPage, IsStable, ResultSetEl, QueryID),
-        ejabberd_sm:route(ArcJID, From, FinMsg),
-
-        %% IQ was sent above
-        ignore;
-    {ok, {TotalCount, Offset, MessageRows}} ->
-        %% Forward messages
-        {FirstMessID, LastMessID} =
-            case MessageRows of
-                []    -> {undefined, undefined};
-                [_|_] -> {message_row_to_ext_id(hd(MessageRows)),
-                          message_row_to_ext_id(lists:last(MessageRows))}
-            end,
-        [send_message(ArcJID, From, message_row_to_xml(MamNs, set_client_xmlns_for_row(M), QueryID))
-         || M <- MessageRows],
-
-        %% Make fin iq
-        IsLastPage = is_last_page(PageSize, TotalCount, Offset, MessageRows),
-        IsStable = true,
-        ResultSetEl = result_set(FirstMessID, LastMessID, Offset, TotalCount),
-        FinElem = make_fin_element(IQ#iq.xmlns, IsLastPage, IsStable, ResultSetEl),
-        IQ#iq{type = result, sub_el = [FinElem]}
+        {error, Reason} ->
+            report_issue(Reason, mam_lookup_failed, ArcJID, IQ),
+            return_error_iq(IQ, Reason);
+        {ok, {TotalCount, Offset, MessageRows}} when IQ#iq.xmlns =:= ?NS_MAM_03 ->
+            ResIQ = IQ#iq{type = result, sub_el = []},
+            %% Server accepts the query
+            ejabberd_router:route(ArcJID, From, jlib:iq_to_xml(ResIQ)),
+
+            %% Forward messages
+            {FirstMessID, LastMessID} =
+                case MessageRows of
+                    []    -> {undefined, undefined};
+                    [_|_] -> {message_row_to_ext_id(hd(MessageRows)),
+                              message_row_to_ext_id(lists:last(MessageRows))}
+                end,
+            [send_message(ArcJID, From,
+                          message_row_to_xml(MamNs, set_client_xmlns_for_row(M), QueryID))
+             || M <- MessageRows],
```

Comment:
Maybe it looked better, but the line was to long. Thats the only reason I've changed the formatting.